src/monodroid/jni/android-system.cc

```diff
@@ -222,13 +213,32 @@ AndroidSystem::_monodroid__system_property_get (const char *name, char *sp_value
 }
 #endif
 
+int
+AndroidSystem::monodroid_get_system_property (const char *name, dynamic_local_string<PROPERTY_VALUE_BUFFER_LEN>& value)
+{
+	int len = _monodroid__system_property_get (name, value.get (), value.size ());
+	if (len > 0) {
+		// Clumsy, but if we want direct writes to be fast, this is the price we pay
+		value.set_length_after_direct_write (static_cast<size_t>(len));
+		return len;
+	}
+
+	size_t plen;
+	const char *v = lookup_system_property (name, plen);
+	if (v == nullptr)
+		return len;
+
+	value.assign (v, plen);
+	return static_cast<int>(plen);
```

Comment:
I think this should be `ADD_WITH_OVERFLOW_CHECK (int, plen, 0)`, assuming that `ADD_WITH_OVERFLOW_CHECK()`/`__builtin_add_overflow()` will reasonably do what I want: ensure that we don't try to return a value larger than `MAX_INT`.